substrate/frame/assets/src/lib.rs

```diff
@@ -1697,7 +1697,9 @@ pub mod pallet {
 
 		fn should_touch(asset: T::AssetId, who: &T::AccountId) -> bool {
 			match Asset::<T, I>::get(&asset) {
+				// refer to the [`Self::new_account`] function for more details.
 				Some(info) if info.is_sufficient => false,
+				Some(_) if frame_system::Pallet::<T>::can_accrue_consumers(who, 2) => false,
```

Comment:
Why does it only touch when it can still accrue consumers? I assume someone could send an asset to this account to use up another consumer just after this was checked here?

Reply:
Yes, someone can, and from that moment, a creation of non sufficient account would require a deposit. The result is only valid for a current transaction. 
Not sure I got the question right. Here the doc for the trait - https://github.com/paritytech/polkadot-sdk/blob/0d71753e0eb1c0fdbeb09af5db7a2e1b5f150b13/substrate/frame/support/src/traits/misc.rs#L1190